### PR TITLE
Create packages

### PR DESCRIPTION
**Pull request by lilian-wang**

### CODE REVIEW

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <tags>webrtc uwp windows native rtc real time communication</tags>
+        <dependencies>
+	    <group targetFramework="native" />
+        </dependencies>
+    </metadata>
+
+    <files>
+        <file src="license.txt"/>
+        <file src="../output/msvc/uwp/Release/x64/obj/webrtc.lib" target="lib/native/Release/x64/v142" />
+        <file src="include/**/*.h" target="build/native" />
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec]: Were the header files copied to the NuGet package?

**Review by LoadLibrary — CHANGES_REQUESTED** [review]
The header files are not being copied correctly.

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <metadata>
+        <id>Microsoft.WinRTC.libwebrtc</id>
+        <version>84.0.3-alpha</version>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec:5]: Is there a way to have this auto generated? (%CDP_PACKAGE_VERSION_NUMERIC%-%CDP_BUILD_TAG%) ?

File: NuSpecs/Microsoft.WinRTC.libwebrtc.props [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?> 
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ImportGroup Label="PropertySheets" />
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup />
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">USE_AURA=1;_HAS_EXCEPTIONS=0;__STD_C;_CRT_RAND_S;_CRT_SECURE_NO_DEPRECATE;_SCL_SECURE_NO_DEPRECATE;_ATL_NO_OPENGL;_WINDOWS;CERT_CHAIN_PARA_HAS_EXTRA_FIELDS;PSAPI_VERSION=2;WIN32;_SECURE_ATL;_USING_V110_SDK71_;WINUWP;__WRL_NO_DEFAULT_LIB__;WINAPI_FAMILY=WINAPI_FAMILY_PC_APP;WIN10=_WIN32_WINNT_WIN10;WIN32_LEAN_AND_MEAN;NOMINMAX;_UNICODE;UNICODE;NTDDI_VERSION=NTDDI_WIN10_RS2;_WIN32_WINNT=0x0A00;WINVER=0x0A00;NDEBUG;NVALGRIND;DYNAMIC_ANNOTATIONS_ENABLED=0;WEBRTC_ENABLE_PROTOBUF=0;WEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE;RTC_ENABLE_VP9;HAVE_SCTP;WEBRTC_USE_H264;RTC_DISABLE_LOGGING;RTC_DISABLE_TRACE_EVENTS;WEBRTC_LIBRARY_IMPL;RTC_DISABLE_CHECK_MSG;WEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0;WEBRTC_WIN;ABSL_ALLOCATOR_NOTHROW=1;HAVE_SCTP;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.props]: m84 changed the required flags to:
USE_AURA=1;_HAS_EXCEPTIONS=0;__STD_C;_CRT_RAND_S;_CRT_SECURE_NO_DEPRECATE;_SCL_SECURE_NO_DEPRECATE;_ATL_NO_OPENGL;_WINDOWS;CERT_CHAIN_PARA_HAS_EXTRA_FIELDS;PSAPI_VERSION=2;WIN32;_SECURE_ATL;WINUWP;__WRL_NO_DEFAULT_LIB__;WINAPI_FAMILY=WINAPI_FAMILY_PC_APP;WIN10=_WIN32_WINNT_WIN10;WIN32_LEAN_AND_MEAN;NOMINMAX;_UNICODE;UNICODE;NTDDI_VERSION=NTDDI_WIN10_RS2;_WIN32_WINNT=0x0A00;WINVER=0x0A00;NDEBUG;NVALGRIND;DYNAMIC_ANNOTATIONS_ENABLED=0;WEBRTC_ENABLE_PROTOBUF=0;WEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE;RTC_ENABLE_VP9;HAVE_SCTP;WEBRTC_LIBRARY_IMPL;WEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0;WEBRTC_WIN;ABSL_ALLOCATOR_NOTHROW=1;HAVE_SCTP;WEBRTC_VIDEO_CAPTURE_WINRT

File: NuSpecs/Microsoft.WinRTC.libwebrtc.props [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">USE_AURA=1;_HAS_EXCEPTIONS=0;__STD_C;_CRT_RAND_S;_CRT_SECURE_NO_DEPRECATE;_SCL_SECURE_NO_DEPRECATE;_ATL_NO_OPENGL;_WINDOWS;CERT_CHAIN_PARA_HAS_EXTRA_FIELDS;PSAPI_VERSION=2;WIN32;_SECURE_ATL;_USING_V110_SDK71_;WINUWP;__WRL_NO_DEFAULT_LIB__;WINAPI_FAMILY=WINAPI_FAMILY_PC_APP;WIN10=_WIN32_WINNT_WIN10;WIN32_LEAN_AND_MEAN;NOMINMAX;_UNICODE;UNICODE;NTDDI_VERSION=NTDDI_WIN10_RS2;_WIN32_WINNT=0x0A00;WINVER=0x0A00;_DEBUG;DYNAMIC_ANNOTATIONS_ENABLED=1;_HAS_ITERATOR_DEBUGGING=0;WEBRTC_ENABLE_PROTOBUF=1;WEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE;RTC_ENABLE_VP9;HAVE_SCTP;WEBRTC_USE_H264;WEBRTC_LIBRARY_IMPL;WEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0;WEBRTC_WIN;ABSL_ALLOCATOR_NOTHROW=1;HAVE_SCTP;GOOGLE_PROTOBUF_NO_RTTI;GOOGLE_PROTOBUF_NO_STATIC_INITIALIZER;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.props]: We are not adding the debug lib file to this package. Feel free to remove this line.

File: NuSpecs/Microsoft.WinRTC.libwebrtc.props [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <AdditionalIncludeDirectories>$(MSBuildThisFileDirectory)include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+    </ClCompile>
+    <Link>
+      <AdditionalDependencies>webrtc.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(MSBuildThisFileDirectory)..\..\lib\native\$(Configuration)\$(Platform)\$(PlatformToolset);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <IgnoreSpecificDefaultLibraries Condition="'$(Configuration)|$(Platform)'=='Release|x64'">libcmt.lib;%(IgnoreSpecificDefaultLibraries)</IgnoreSpecificDefaultLibraries>
+      <IgnoreSpecificDefaultLibraries Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">libcmtd.lib;%(IgnoreSpecificDefaultLibraries)</IgnoreSpecificDefaultLibraries>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.props]: We're not adding the debug lib to this package. Feel free to remove this line.

File: .pipelines/build.cmd [hunk 6]
```diff
@@ -33,7 +33,26 @@ echo Copying contents...
...
+xcopy /s /i c:\webrtc\src\p2p\*.h ..\include\p2p\
+xcopy /s /i c:\webrtc\src\pc\*.h ..\include\pc\
+xcopy /s /i c:\webrtc\src\rtc_base\*.h ..\include\rtc_base\
+xcopy /s /i c:\webrtc\src\rtc_tools\*.h ..\include\rtc_tools\
+xcopy /s /i c:\webrtc\src\sdk\*.h ..\include\sdk\
+xcopy /s /i c:\webrtc\src\stats\*.h ..\include\stats\
+xcopy /s /i c:\webrtc\src\system_wrappers\*.h ..\include\system_wrappers\
+xcopy /s /i c:\webrtc\src\video\*.h ..\include\video\
+xcopy /s /i c:\webrtc\src\common_types.h ..\include\
 if errorlevel 1 goto :error
```
Comment (LoadLibrary) [inline .pipelines/build.cmd]: How about checking for error on each xcopy command? Or creating a function taking a list as parameter and checking return code?

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        <description>Core utility functions for web applications</description>
+        <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+        <tags>webrtc uwp windows native rtc real time communication</tags>
+        <dependencies>
+	    <group targetFramework="native" />
+        </dependencies>
+    </metadata>
+
+    <files>
+        <file src="license.txt"/>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec]: nit: space before the forward slash

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec (hunk 2 above)
Comment (lilian-wang) [reply NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec:5]: Yes, the package.cmd file overrides the version.

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <metadata>
+        <id>Microsoft.WinRTC.libwebrtc</id>
+        <version>84.0.3-alpha</version>
+        <authors>Microsoft</authors>
+        <owners>Microsoft</owners>
+        <projectUrl>http://github.com/microsoft/winrtc</projectUrl>
+        <license type="file">license.txt</license>
+        <requireLicenseAcceptance>true</requireLicenseAcceptance>
+        <releaseNotes>First drop</releaseNotes>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec]: Please replace with: Upgrading WinRTC with the WebRTC changes up to version m84. Bringing the Media Foundation based H264 encoder and decoder from the WebRTC-UWP project.

File: NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        <description>Core utility functions for web applications</description>
```
Comment (LoadLibrary) [inline NuSpecs/Microsoft.WinRTC.libwebrtc.nuspec]: Please replace with: This package contains a WebRTC static library and the header files to be used to build an UWP native app or component.

File: .pipelines/build.cmd [hunk 10]
```diff
@@ -33,9 +33,32 @@ echo Copying contents...
...
+call :copy c:\webrtc\src\rtc_tools\*.h ..\include\rtc_tools\
+call :copy c:\webrtc\src\sdk\*.h ..\include\sdk\
+call :copy c:\webrtc\src\stats\*.h ..\include\stats\
+call :copy c:\webrtc\src\system_wrappers\*.h ..\include\system_wrappers\
+call :copy c:\webrtc\src\video\*.h ..\include\video\
+call :copy c:\webrtc\src\common_types.h ..\include\
+
+goto :exit
 
+:copy
```
Comment (LoadLibrary) [inline .pipelines/build.cmd]: The name of this function is confusing. There is a cmd command named `copy`. How about renaming this function to `copyHeaders` ?

File: .pipelines/build.cmd [hunk 11]
```diff
@@ -33,9 +33,32 @@ echo Copying contents...
...
-xcopy /e /i c:\webrtc\src\out ..\output
-if errorlevel 1 goto :error
+call :copy c:\webrtc\src\out\msvc\uwp\Release\x64\obj\webrtc.lib ..\output\msvc\uwp\Release\x64\obj\
+call :copy c:\webrtc\src\api\*.h ..\include\api\
+call :copy c:\webrtc\src\audio\*.h ..\include\audio\
+call :copy c:\webrtc\src\base\*.h ..\include\base\
+call :copy c:\webrtc\src\call\*.h ..\include\call\
+call :copy c:\webrtc\src\common_audio\*.h ..\include\common_audio\
+call :copy c:\webrtc\src\common_video\*.h ..\include\common_video\
+call :copy c:\webrtc\src\data\*.h ..\include\data\
```
Comment (LoadLibrary) [inline .pipelines/build.cmd]: Please remove this line. The [log file](https://github-private.visualstudio.com/microsoft/_build/results?buildId=7866&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=dcf93f35-c1ba-5dce-0078-8d628250da3a&l=3789) reports that there is no header files in the `data` folder.
```
[INFO] [3720] C:\webrtc\src\common_video\test\utilities.h
[INFO] [3721] 17 File(s) copied
[INFO] [3722] 0 File(s) copied
##[warning][Build] [Build WebRTC] [3723] File not found - *.h
[WARN] [3723] File not found - *.h
[INFO] [3724] C:\webrtc\src\logging\rtc_event_log\fake_rtc_event_log.h
```

File: .pipelines/build.cmd [hunk 12]
```diff
@@ -33,9 +33,32 @@ echo Copying contents...
...
+call :copy c:\webrtc\src\logging\*.h ..\include\logging\
+call :copy c:\webrtc\src\media\*.h ..\include\media\
+call :copy c:\webrtc\src\modules\*.h ..\include\modules\
+call :copy c:\webrtc\src\p2p\*.h ..\include\p2p\
+call :copy c:\webrtc\src\pc\*.h ..\include\pc\
+call :copy c:\webrtc\src\rtc_base\*.h ..\include\rtc_base\
+call :copy c:\webrtc\src\rtc_tools\*.h ..\include\rtc_tools\
+call :copy c:\webrtc\src\sdk\*.h ..\include\sdk\
```
Comment (LoadLibrary) [inline .pipelines/build.cmd]: We don't need the `sdk` as well. This is just for android and iOS.

File: .pipelines/build.cmd (hunk 10 above)
Comment (lilian-wang) [reply .pipelines/build.cmd]: Renamed to copyFiles. We are also copying .lib file.

File: .pipelines/build.cmd [hunk 13]
```diff
@@ -33,9 +33,30 @@ echo Copying contents...
...
+call :copyFiles c:\webrtc\src\media\*.h ..\include\media\
+call :copyFiles c:\webrtc\src\modules\*.h ..\include\modules\
+call :copyFiles c:\webrtc\src\p2p\*.h ..\include\p2p\
+call :copyFiles c:\webrtc\src\pc\*.h ..\include\pc\
+call :copyFiles c:\webrtc\src\rtc_base\*.h ..\include\rtc_base\
+call :copyFiles c:\webrtc\src\rtc_tools\*.h ..\include\rtc_tools\
+call :copyFiles c:\webrtc\src\stats\*.h ..\include\stats\
+call :copyFiles c:\webrtc\src\system_wrappers\*.h ..\include\system_wrappers\
+call :copyFiles c:\webrtc\src\video\*.h ..\include\video\
+call :copyFiles c:\webrtc\src\common_types.h ..\include\
```
Comment (LoadLibrary) [inline .pipelines/build.cmd:55]: Please add abseil and libyuv:

xcopy /s /i c:\webrtc\src\third_party\abseil-cpp\absl\*.h C:\WinRTC_GitHub\MyFirstWinRtc\packages\Microsoft.Win
RTC.libwebrtc.84.0.12980001-alpha\build\native\include\absl\

xcopy /s /i c:\webrtc\src\third_party\libyuv\include\*.h C:\WinRTC_GitHub\MyFirstWinRtc\packages\Microsoft.
WinRTC.libwebrtc.84.0.12980001-alpha\build\native\include\

File: .pipelines/build.cmd [hunk 14]
```diff
@@ -33,9 +33,30 @@ echo Copying contents...
 cd /D "%~dp0"
 if errorlevel 1 goto :error
 
-xcopy /e /i c:\webrtc\src\out ..\output
-if errorlevel 1 goto :error
+call :copyFiles c:\webrtc\src\out\msvc\uwp\Release\x64\obj\webrtc.lib ..\output\msvc\uwp\Release\x64\obj\
+call :copyFiles c:\webrtc\src\api\*.h ..\include\api\
```
Comment (LoadLibrary) [inline .pipelines/build.cmd:37]: Please make sure the files are copied into the "include" folder.

**Review by LoadLibrary — APPROVED** [review]
I've downloaded the resulting .nuget file and I was build a test app. Ship it!

**Pull request merged**